### PR TITLE
Config improve

### PR DESCRIPTION
**Pull request by x-ror**
Improved rollup.config.js for ESM and CJS format. 
move dist files to `/providers/index.{m}js`,  `/client/index.{m}js` and `./index.{m}js`
used svelte style for rollup config and `import/require` in `package.json`
used `prepare` command so now package is packed and published and on local npm install

### CODE REVIEW

**Comment by Dan6erbond** [comment]
I'll have to give this a look. Seeing the way the bundle is made adds a lot of clutter to the file structure, and if that can be avoided I'll keep things the way they are. At the moment the build seems to work on a variety of setups, so as long as there are no issues I prefer the current way we're doing things.

**Comment by vhscom** [comment]
@Dan6erbond I saw you flagged this Help Wanted. I'm available to provide assistance. Regarding the build working on a variety of setups is that really necessary? Since this is SvelteKit basted I'd expect a single ESM target to be sufficient. Am I missing something?

**Comment by vhscom** [comment]
@Dan6erbond I saw you flagged this Help Wanted. I'm available to provide assistance. Regarding the build working on a variety of setups is that really necessary? Since this is SvelteKit based I'd expect a single ESM target to be sufficient. Am I missing something?

**Comment by vhscom** [comment]
@Dan6erbond I saw you flagged this Help Wanted. I'm available to provide assistance. Regarding the build working on a variety of setups is that really necessary? Since this is SvelteKit based I'd expect a single ESM target to be sufficient. Is the thinking that this will not work when packaged up in an app that modifies their `tsconfig` settings?

**Comment by vhscom** [comment]
@Dan6erbond I saw you flagged this Help Wanted. I'm available to provide assistance. Regarding the build working on a variety of setups is that really necessary? Since this is SvelteKit based I'd expect a single ESM target to be sufficient. Is the thinking that this library will no longer work when packaged up in an app that modifies their `tsconfig` settings?

File: package.json [hunk 1]
```diff
@@ -2,14 +2,23 @@
   "name": "sk-auth",
   "version": "0.3.7",
   "description": "Authentication library for use with SvelteKit featuring built-in OAuth providers and zero restriction customization!",
-  "main": "dist/index.js",
-  "module": "dist/index.esm.js",
-  "types": "dist/index.d.ts",
+  "module": "index.mjs",
```
Comment (vhscom) [inline package.json:5]: I'm sure I'm not alone when I say `.mjs` files were one of the silliest decisions the Node community made. What they should've done was used `.njs` for anything which required a Node runtime to function.

Comment (vhscom) [inline package.json:5]: I'm sure I'm not alone when I say suggesting `.mjs` file extensions was one of the silliest proposals the Node maintainers have made. That would be like proposing `.njs` for anything which required a Node runtime. Only the latter makes more sense. Note with #78 the Node runtime is no longer needed for this library to function.

Comment (Dan6erbond) [reply package.json:5]: I personally do not like this addition either, which is why I'll be closing it up now. Thanks for the review @vhscom!

**Pull request closed without merging**